### PR TITLE
Feature 291 add black formatting as ci

### LINKED ISSUE #291: Add black formatting as CI
Should we add a black formatting github action like described in this video (at 6:36) https://www.youtube.com/watch?v=j1MbEYhYj_Y

### PR DESCRIPTION
**Pull request by deniztepe**
The pre-commit hook is activated, after pip installing the `pre-commit` package and running `pre-commit install` on console. You can test it on the feature branch, if you want to. I also added a short documentation of the feature.

### CODE REVIEW

**Comment by FlorianK13** [comment]
I have adapted the flake8 file to be black compatible. There is still the pylint file (.pylintrc) in the main directory which is also for code formatting. Can we delete it since we only use black + flake8 ?

**Comment by deniztepe** [comment]
I would say, yes?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+      - id: black
+        language_version: python3.9
```
Comment (FlorianK13) [inline .pre-commit-config.yaml:6]: what does python3.9 mean here? Can it have conflicts with people using other versions?

Comment (deniztepe) [reply .pre-commit-config.yaml:6]: It was on the default template. As far as I understood, it determines the python version for the hook. Black is called with this version. I assume it should be compatible with other versions such as 3.7...3-10.

Comment (FlorianK13) [reply .pre-commit-config.yaml:6]: Let's merge it then and have a bugfix if it does not work.

**Review by FlorianK13 — APPROVED** [review]

**Pull request merged**